crates/optimism/chainspec/src/lib.rs

```diff
@@ -178,6 +188,51 @@ pub struct OpChainSpec {
     pub inner: ChainSpec,
 }
 
+/// Fee trait for OP chain specs.
+pub trait Fee {
+    /// Read from parent to determine the base fee for the next block
+    fn next_block_base_fee(&self, parent: &Header, timestamp: u64) -> U256;
+}
```

Comment:
does this need to be a trait / do we expect alternate implementations of `Fee`? Or should these actually just be implemented as regular methods on `OpChainSpec`